tests/test_globus_compute_mpi.py

```diff
@@ -0,0 +1,419 @@
+            compile_func,
+            env=config["environment"],
+            dirpath=pwd,
+        )
+        r = future.result()
+        assert r.returncode == 0
+
+    # Run the MPI program in the MPI endpoint
+    with Executor(endpoint_id=mpi_endpoint_id) as gce:
+        gce.resource_specification = {
+            "num_nodes": 3,  # Number of nodes required for the application instance
+            "num_ranks": 6,  # Number of ranks in total
+            "ranks_per_node": 2,  # Number of ranks / application elements to be launched per node
+        }
+        future = gce.submit(
+            hello_func,
+            env=config["environment"],
+            dirpath=pwd,
+        )
+        r = future.result()
+        assert r.returncode == 0
+
+    # Check output
+    with open(pwd / "globus_compute_mpi_hello_run.out", "r") as f:
+        for line in f:
+            assert re.match(r"Hello world from host \S+, rank \d+ out of 6", line)
+
+
+# Test endpoint use
+def test_endpoint_mpi_pi(config):
+    pwd = pathlib.Path(__file__).parent.resolve()
+
+    # Define a ShellFunction to compile the MPI code
+    compile_func = ShellFunction(
+        """
+        {env}
+        cd {dirpath}
+        $CHILTEPIN_MPIF90 -o mpi_pi.exe mpi_pi.f90
+        """,
+        stdout=os.path.join(pwd, "globus_compute_mpi_pi_compile.out"),
+        stderr=os.path.join(pwd, "globus_compute_mpi_pi_compile.err"),
+    )
+
+    # Define a MPIFunction to run the MPI program
+    pi1_func = MPIFunction(
+        """true;  # Force the default prefix to launch a no-op
+        {env}
+        cd {dirpath}
+        $PARSL_MPI_PREFIX --overcommit ./mpi_pi.exe
+        """,
+        stdout=os.path.join(pwd, "globus_compute_mpi_pi1_run.out"),
+        stderr=os.path.join(pwd, "globus_compute_mpi_pi1_run.err"),
+    )
+
+    # Define a MPIFunction to run the MPI program
+    pi2_func = MPIFunction(
+        """true;  # Force the default prefix to launch a no-op
+        {env}
+        cd {dirpath}
+        $PARSL_MPI_PREFIX --overcommit ./mpi_pi.exe
+        """,
+        stdout=os.path.join(pwd, "globus_compute_mpi_pi2_run.out"),
+        stderr=os.path.join(pwd, "globus_compute_mpi_pi2_run.err"),
+    )
+
+    # Get a listing of the endpoints
+    p = subprocess.run(
+        [
+            "globus-compute-endpoint",
+            "-c",
+            f"{pwd}/globus_compute",
+            "list",
+        ],
+        stdout=subprocess.PIPE,
+        stderr=subprocess.STDOUT,
+        text=True,
+        timeout=60,
+    )
+    assert p.returncode == 0
+
+    # Get the uuid of the mpi endpoint
+    mpi_endpoint_regex = re.compile(r"\| ([0-9a-f\-]{36}) \| Running\s+\| mpi\s+\|")
+    match = mpi_endpoint_regex.search(p.stdout)
+    mpi_endpoint_id = match.group(1)
+    assert len(mpi_endpoint_id) == 36
+
+    # Get the uuid of the compute endpoint
+    compute_endpoint_regex = re.compile(
+        r"\| ([0-9a-f\-]{36}) \| Running\s+\| compute\s+\|"
+    )
```

Comment:
Can you filter some of this out to be reused? In the test above it also calls the same information as here.

Reply:
I pushed a fix for this.  Please take a look.